service/storage_service.cc

```diff
@@ -2928,21 +2928,23 @@ future<> storage_service::removenode_with_stream(gms::inet_address leaving_node,
 }
 
 future<> storage_service::restore_replica_count(inet_address endpoint, inet_address notify_endpoint) {
+    _abort_source.check();
+    auto sas = make_shared<abort_source>();
+    auto sub = _abort_source.subscribe([sas] () noexcept {
```

Comment:
Do we really need a shared_ptr here? Usually the owner of the abort_source outlives the aborted thing.

Reply:
We currently need a `shared_ptr<abort_source>` for `node_ops_info`.
This will change in https://github.com/scylladb/scylladb/pull/12085

Reply:
Ok. But we should strive for nesting lifetimes, not overlapping them. This makes for fewer bugs.